fesh/fesh

```diff
@@ -30,48 +30,74 @@ set -euo pipefail
 
 FS_BASE=${FS_BASE:-'/usr2'}
 FS_PATH=${FS_PATH:-"${FS_BASE}/fs"}
+CONTROL_PATH=${CONTROL_PATH:-"${FS_BASE}/control"}
 LOG_PATH=${LOG_PATH:-"${FS_BASE}/log"}
-PROC_PATH=${PROC_PATH:-"${FS_BASE}/proc"}
-SCHED_PATH=${SCHED_PATH:-"${FS_BASE}/sched"}
-SNAP_PATH=${SNAP_PATH:-"${FS_BASE}/sched"}
+# check for skedf.ctl
+if [[ -e $CONTROL_PATH/skedf.ctl ]]; then
+    proc_path=$( perl -ne 'END {if(eof) {$_=".";} print;} if(/^\$proc/i     ) {while (<>){if(/^\$/) {$_=".";exit;} elsif(/^[ \t]*$/) {}elsif(/^[ \t]*([^* \t].*?)\/* *$/) {$_=$1;exit;}}}' <$CONTROL_PATH/skedf.ctl)
+    snap_path=$( perl -ne 'END {if(eof) {$_=".";} print;} if(/^\$snap/i     ) {while (<>){if(/^\$/) {$_=".";exit;} elsif(/^[ \t]*$/) {}elsif(/^[ \t]*([^* \t].*?)\/* *$/) {$_=$1;exit;}}}' <$CONTROL_PATH/skedf.ctl)
+    sched_path=$(perl -ne 'END {if(eof) {$_=".";} print;} if(/^\$schedules/i) {while (<>){if(/^\$/) {$_=".";exit;} elsif(/^[ \t]*$/) {}elsif(/^[ \t]*([^* \t].*?)\/* *$/) {$_=$1;exit;}}}' <$CONTROL_PATH/skedf.ctl)
```

Comment:
ooph. I wonder if this could be put into a function, so at least those of us that have trouble reading Perl even with the new lines might have a chance :)

Reply:
:) But is so compact. It was a  little better before I had to add the despised middle `if` to catch empty lines (still don't understand that" somehow it thinks there was a match even if one field didn't match?).

I tried a function. Although I am getting better with `bash`, I couldn't figure out quickly how to quote it properly to allow an argument to be passed in as well. Also, the extra quoting characters made it even more unreadable IMNSHO. Would Here Document work better? I can try that if you think so. Three functions seemed like a loser. There is the question of how much to pretty this up. This is a just a stop-gap (I guess) until `fesh2`.